### PR TITLE
fix getNeighbor iterator

### PR DESCRIPTION
**Pull request by nevermore3**
Add rows without edge to noEdgeRows_ , 
Then Subgraph can collect the vertices of each step for the getNeigbors

### CODE REVIEW

**Comment by nevermore3** [comment]
D8B4924B-D10E-4171-9069-7E8ED5AAF055
we should add  (Pacers) and (Thunders) to logicalRow

File: src/context/Iterator.cpp [hunk 1]
```diff
@@ -96,6 +96,7 @@ StatusOr<GetNeighborsIter::DataSetIndex> GetNeighborsIter::makeDataSetIndex(cons
 void GetNeighborsIter::makeLogicalRowByEdge(int64_t edgeStartIndex,
                                             size_t idx,
                                             const DataSetIndex& dsIndex) {
+    bool existEdge = false;
     for (auto& row : dsIndex.ds->rows) {
         auto& cols = row.values;
```
Comment (yixinglu) [inline src/context/Iterator.cpp:101]: Init `existEdge` variable here

File: src/context/Iterator.cpp [hunk 2]
```diff
@@ -108,12 +109,17 @@ void GetNeighborsIter::makeLogicalRowByEdge(int64_t edgeStartIndex,
...
                 auto edgeName = dsIndex.tagEdgeNameIndices.find(column);
                 DCHECK(edgeName != dsIndex.tagEdgeNameIndices.end());
                 logicalRows_.emplace_back(
                     idx, &row, edgeName->second, &edge.getList());
             }
         }
+        if (!existEdge) {
+            logicalRows_.emplace_back(idx, &row, "", nullptr);
+        }
+        existEdge = false;
```
Comment (yixinglu) [inline src/context/Iterator.cpp:122]: And need not it

**Review by yixinglu — COMMENTED** [review]
Could you add some test cases for this scenario ?

File: src/context/Iterator.cpp (hunk 2 above)
Comment (nevermore3) [reply src/context/Iterator.cpp]: done

File: src/context/Iterator.cpp (hunk 1 above)
Comment (nevermore3) [reply src/context/Iterator.cpp:101]: done

**Comment by nevermore3** [comment]
> Could you add some test cases for this scenario ?

done

**Review by CPWstatic — APPROVED** [review]

**Review by Shylock-Hg — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Pull request merged**